service/history/timerQueueActiveProcessor.go

```diff
@@ -369,6 +371,11 @@ func (t *timerQueueActiveProcessorImpl) processActivityTimeout(
 		return err
 	}
 	if mutableState == nil || !mutableState.IsWorkflowExecutionRunning() {
+		t.metricsClient.RecordTimer(
```

Comment:
Why record the timer here?

Reply:
It's possible the function will return here without any error. so emit a metric to see the latency.